### DISCUSSION TITLE
Share Substrate Mobile Engineering Tools / Libraries

### DISCUSSION THREAD

**Discussion by hanwencheng** [paritytech/substrate-open-working-groups#16] [Working Groups]
As the curator of the Kusama [Bounty #3](https://kusama.polkassembly.io/bounty/3) and the developer of Parity Signer, I have talked to several teams about their contributions to the Polkadot/Kusama ecosystem and have seen many libraries/tools have been built by these teams. It would be great if we could list all the tools/libraries related to Substrate-based blockchain, and further discuss what we still need in Substrate mobile engineering.

I hope this thread could provide useful information, prevent duplication work, and raise new ideas for different substrate mobile engineering teams.

|name | description|Language| Github repo|
|----|----|----|----|
|Polkadot.js API| Well-known Javascript library for Substrate| Typescript |https://github.com/polkadot-js/api|
|React-Native-Substrate-Sign|Rust library for React Native| Rust, Javascript |https://github.com/paritytech/react-native-substrate-sign|
|Polkadot-Dart| Dart Substrate API| Dart|  https://github.com/Pocket4D/Polkadot-Dart|
|PolkaWallet SDK| Flutter SDK for Substrate-based App|Dart, Typescript |https://github.com/polkawallet-io/sdk|
|Fearless Utils iOS| iOS Substrate tools| Swtift |https://github.com/soramitsu/fearless-utils-iOS|
|Fearless Utils Android | Android Substrate tools | Kotlin | https://github.com/soramitsu/fearless-utils-Android|

Feel free to comment below if you want to add more items.

**Comment by BenWhiteJam** [comment]
@NukeManDan where do you think would be a good place for this to get added?

**Reply by jimmychu0807** [reply]
Since these projects are built and maintained by the community, how about merging it to the [Awesome Substrate list](https://github.com/substrate-developer-hub/awesome-substrate)?

**Reply by nuke-web3** [reply]
Totally agree! @hanwencheng would you be open to a PR on awesome substrate to include these? Likely best to use the current sections, with mobile / iOS/android mentioned in their descriptions

**Reply by jimmychu0807** [reply]
@hanwencheng @BenWhiteJam Added a PR putting the above content in Awesome Substrate list:
https://github.com/substrate-developer-hub/awesome-substrate/pull/49
